serverjs/util.js

```diff
@@ -112,6 +112,31 @@ function fromEntries(entries) {
   return obj;
 }
 
+async function addNotification(user, from, url, text) {
+  if (user._id == from._id) {
+    return; //we don't need to give notifications to ourselves
+  }
+  user.notifications.push({
+    user_from: from._id,
+    user_from_name: from.username,
+    url: url,
+    date: new Date(),
+    text: text
+  });
+  user.old_notifications.push({
+    user_from: from._id,
+    user_from_name: from.username,
+    url: url,
+    date: new Date(),
+    text: text
+  });
+  if (user.old_notifications.length > 200) {
```

Comment:
I would change this to a while loop just to avoid any weird conditions where the array ends up with more notifications than it should have